sonic-chassisd/scripts/chassisd

```diff
@@ -393,7 +393,7 @@ class ModuleUpdater(logger.Logger):
 
         module_info_dict[CHASSIS_MODULE_INFO_NAME_FIELD] = name
         module_info_dict[CHASSIS_MODULE_INFO_DESC_FIELD] = str(desc)
-        module_info_dict[CHASSIS_MODULE_INFO_SLOT_FIELD] = str(slot)
+        module_info_dict[CHASSIS_MODULE_INFO_SLOT_FIELD] = slot
```

Comment:
This may be an unexpected behavior change. `slot` can still be an integer, so `module_info_dict[CHASSIS_MODULE_INFO_SLOT_FIELD]` can now have varying types. Previously when `slot was always `int`, all logic was expecting this field to be `str` since it was converted here (though I'm not sure any logic relies on that type). It may be best to revert this change to maintain a consistent (and existing) type requirement.

I do see there is a conversion for one of the usages of this dict entry to `str` to account for this, but it may just be best to maintain the expectation that this dict entry is always `str` from this point, as it was previously.

Reply:
The fundamental issue is a slot for chassis my not be number,  it could be a string -- such as Nokia IXRE7250 
It is not necessary to convert to string in the buffer here.  We can directly use it as what function get_slot() returned.  The only place is needed to be converted to a string when it is stored into a STATE_DB.  This change simply and no need to  convert back and forth (StringToNumber or NumberToString) when use the slot info in module_info_dict[].

Reply:
Sounds good. My concern was mostly that it may be unclear to developers that the type stored in this dict field can vary. Functionally it seems fine. The documentation of APIs such as `get_slot` and `get_my_slot` should be updated as @judyjoseph previously mentioned; they not need not return only an int, they may also return str.

Reply:
> Sounds good. My concern was mostly that it may be unclear to developers that the type stored in this dict field can vary. Functionally it seems fine. The documentation of APIs such as `get_slot` and `get_my_slot` should be updated as @judyjoseph previously mentioned; they not need not return only an int, they may also return str.

Agree.  For the long run, this needs to be addressed.